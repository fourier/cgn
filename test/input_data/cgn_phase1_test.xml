<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn" 
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java">
  <!-- this test tests the 1st transformation phase:
       verification of the predefined attribute values
       correctness and transformation them to the
       same form
  -->

  <!-- verifying the following attributes:
       cgn:read-only
       cgn:json
       cgn:is-set
       possible values: case-insensitive true/false or yes/no
       all other values generate warning and interpreted as false
  -->

  <!-- verifying on a objects level, input: invalid -->
  <objects cgn:package="com.test.phase1.objects.level.wrong"
           cgn:read-only="don't know"
           cgn:json="not sure"
           cgn:is-set="what to set">
    <object cgn:name="Test1">
      <field cgn:name="some-field"/>
    </object>
  </objects>

  <!-- verifying on a objects level, input: case-insensitive -->
  <objects cgn:package="com.test.phase1.objects.level.true.false"
           cgn:read-only="True"
           cgn:json="False"
           cgn:is-set="No">
    <object cgn:name="Test1">
      <field cgn:name="some-field"/>
    </object>
  </objects>

  <!-- verifying on a object level, input: invalid -->
  <objects cgn:package="com.test.phase1.object.level.wrong">
    <object cgn:name="Test1" cgn:read-only="don't know">
      <field cgn:name="some-field"/>
    </object>
    <object cgn:name="Test2" cgn:json="not sure">
      <field cgn:name="some-field"/>
    </object>
    <object cgn:name="Test3" cgn:is-set="what to set">
      <field cgn:name="some-field"/>
    </object>

  </objects>

  <!-- verifying on a objects level, input: case-insensitive -->
  <objects cgn:package="com.test.phase1.object.level.true.false">
    <object cgn:name="Test1" cgn:read-only="True">
      <field cgn:name="some-field"/>
    </object>
    <object cgn:name="Test2" cgn:json="False">
      <field cgn:name="some-field"/>
    </object>
    <object cgn:name="Test3" cgn:is-set="yEs">
      <field cgn:name="some-field"/>
    </object>
    <object cgn:name="Test4" cgn:is-set="nO">
      <field cgn:name="some-field"/>
    </object>

  </objects>

  

</root>

