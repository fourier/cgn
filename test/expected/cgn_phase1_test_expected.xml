<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn"
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java"><!-- this test tests the 1st transformation phase:
       verification of the predefined attribute values
       correctness and transformation them to the
       same form
  --><!-- verifying the following attributes:
       cgn:read-only
       cgn:json
       cgn:is-set
       possible values: case-insensitive true/false or yes/no
       all other values generate warning and interpreted as false
  --><!-- verifying on a objects level, input: invalid -->
   <objects cgn:package="com.test.phase1.objects.level.wrong"
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false">
      <object cgn:name="Test1">
         <field cgn:name="some-field"/>
      </object>
   </objects>
   <!-- verifying on a objects level, input: case-insensitive -->
   <objects cgn:package="com.test.phase1.objects.level.true.false"
            cgn:read-only="true"
            cgn:json="false"
            cgn:is-set="false">
      <object cgn:name="Test1">
         <field cgn:name="some-field"/>
      </object>
   </objects>
   <!-- verifying on a object level, input: invalid -->
   <objects cgn:package="com.test.phase1.object.level.wrong">
      <object cgn:name="Test1" cgn:read-only="false">
         <field cgn:name="some-field"/>
      </object>
      <object cgn:name="Test2" cgn:json="false">
         <field cgn:name="some-field"/>
      </object>
      <object cgn:name="Test3" cgn:is-set="false">
         <field cgn:name="some-field"/>
      </object>
   </objects>
   <!-- verifying on a objects level, input: case-insensitive -->
   <objects cgn:package="com.test.phase1.object.level.true.false">
      <object cgn:name="Test1" cgn:read-only="true">
         <field cgn:name="some-field"/>
      </object>
      <object cgn:name="Test2" cgn:json="false">
         <field cgn:name="some-field"/>
      </object>
      <object cgn:name="Test3" cgn:is-set="true">
         <field cgn:name="some-field"/>
      </object>
      <object cgn:name="Test4" cgn:is-set="false">
         <field cgn:name="some-field"/>
      </object>
   </objects>
</root>
